### PR TITLE
APP-37: setup linting, typechecking, commit hooks, etc.

### PR DESCRIPTION
**Pull request by NoahCardoza**
Backend
- mypy
- isort
- autopep8

Frontend
- eslint
- prettier
- husky

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -13,8 +13,7 @@ dist-ssr
 *.local
 
 # Editor directories and files
-.vscode/*
-!.vscode/extensions.json
+.vscode
```
Comment (mruiz42) [inline frontend/.gitignore:16]: Which gitignore gets respected by git? Project root or subdirectories or both? Should we be maintaining 3 gitignore files?

**Review by mruiz42 — APPROVED** [review]
lgtm

File: frontend/.gitignore (hunk 1 above)
Comment (NoahCardoza) [reply frontend/.gitignore:16]: Not sure... they were already there. There isn't one in the backend I don't think. I'll merge them all into one in the root.

**Review by AlpyneDreams — APPROVED** [review]
lgtm

**Pull request merged**